File: ChiliCookoff2/Models/PartyDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace ChiliCookoff2.Models
{
    public class PartyDTO : IValidatableObject
    {
        public Guid PartyId { get; set; }
        [Required]
        public string PartyName { get; set; }
        public DateTime? PartyDate { get; set; }
        public bool? EnterChili { get; set; }
        public string ChiliName { get; set; }
        public string HostName { get; set; }
        public int? ChiliNumber { get; set; }
        public string PartyCode { get; set; }
        public int MaxChilis { get; set; }
        public bool HasVotingStarted { get; set; }
        public bool HaveVotesBeenTallied { get; set; }
        public bool HasSubmittedBallot { get; set; }
        public bool IsHostUpdate { get; set; }

        IEnumerable<ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            if (EnterChili.HasValue && EnterChili.Value && string.IsNullOrEmpty(ChiliName))
                yield return new ValidationResult("Chili Name is required.");
        }
    }
}

File: ChiliCookoff2/src/app/PartyDTO.ts
export class PartyDTO {
    public PartyId: string;
    public PartyName: string;
    public PartyDate: Date;
    public EnterChili: boolean;
    public ChiliName: string;
    public HostName: string;
    public ChiliNumber: number;
    public PartyCode: string;
    public MaxChilis: number;
    public HasVotingStarted: boolean;
    public HaveVotesBeenTallied: boolean;
    public HasSubmittedBallot: boolean;
    public IsHostUpdate: boolean;

    constructor(
        PartyId: string,
        PartyName: string,
        PartyDate: Date,
        EnterChili: boolean,
        ChiliName: string,
        HostName: string,
        ChiliNumber: number,
        PartyCode: string,
        MaxChilis: number,
        HasVotingStarted: boolean,
        HaveVotesBeenTallied: boolean,
        HasSubmittedBallot: boolean,
        IsHostUpdate: boolean
    ) {
        this.PartyId = PartyId;
        this.PartyName = PartyName;
        this.PartyDate = PartyDate;
        this.EnterChili = EnterChili;
        this.ChiliName = ChiliName;
        this.HostName = HostName;
        this.ChiliNumber = ChiliNumber;
        this.PartyCode = PartyCode;
        this.MaxChilis = MaxChilis;
        this.HasVotingStarted = HasVotingStarted;
        this.HaveVotesBeenTallied = HaveVotesBeenTallied;
        this.HasSubmittedBallot = HasSubmittedBallot;
        this.IsHostUpdate = IsHostUpdate;
    }
}